src/Dropdown.php

```diff
@@ -4,359 +4,696 @@
+     * as a normal-sized button.
+     *
+     * @return self A new instance with the specified dropdown toggle button size small setting.
      */
-    public function submenuOptions(array $value): self
+    public function toggleSizeSmall(bool $value = true): self
     {
         $new = clone $this;
-        $new->submenuOptions = $value;
+        $new->toggleSize = $value === true ? 'btn-sm' : null;
 
         return $new;
     }
 
     /**
-     * @param array $value the HTML attributes for the widget container tag. The following special options are
-     * recognized.
+     * Whether to render the dropdown toggle button as a split button.
+     *
+     * @param bool $value Whether to render the dropdown toggle button as a split button. If set to `true`, the dropdown
+     * toggle button will be rendered as a split button. If set to `false`, the dropdown toggle button will be rendered
+     * as a normal button.
      *
-     * {@see Html::renderTagAttributes()} for details on how attributes are being rendered.
+     * @return self A new instance with the specified dropdown toggle button split setting.
      */
-    public function options(array $value): self
+    public function toggleSplit(bool $value = true): self
     {
         $new = clone $this;
-        $new->options = $value;
-        $new->tagName = null;
+        $new->toggleSplit = $value;
 
         return $new;
     }
 
     /**
-     * Options for each item if not present in self
+     * Sets the content of the dropdown toggle split button.
+     *
+     * @param string|Stringable $value The content of the dropdown toggle split button.
+     *
+     * @return self A new instance with the specified content of the dropdown toggle split button.
      */
-    public function itemOptions(array $options): self
+    public function toggleSplitContent(string|Stringable $value): self
     {
         $new = clone $this;
-        $new->itemOptions = $options;
+        $new->toggleSplitContent = (string) $value;
 
         return $new;
     }
 
     /**
-     * Options for each item link if not present in current item
+     * Sets the URL for the dropdown toggle link.
+     *
+     * @param string $value The URL for the dropdown toggle link.
+     *
+     * @return self A new instance with the specified URL for the dropdown toggle link.
      */
-    public function linkOptions(array $options): self
+    public function toggleUrl(string $value): self
     {
         $new = clone $this;
-        $new->linkOptions = $options;
+        $new->toggleUrl = $value;
 
         return $new;
     }
 
     /**
-     * Renders menu items.
+     * Sets the variant for the dropdown toggle button. The following options are allowed:
      *
-     * @param array $items the menu items to be rendered
+     * - `DropdownToggleVariant::PRIMARY`: The primary variant.
+     * - `DropdownToggleVariant::SECONDARY`: The secondary variant.
```

Comment:
Full description could stay in an enum only.

Reply:
done.